envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/SnapshotProperties.kt

```diff
@@ -249,6 +250,10 @@ class EgressProperties {
 
 class IngressProperties {
     var headersToRemove = mutableListOf<String>()
+    var addServiceNameHeaderToResponse = false
+    var addRequestedServiceNameHeaderToResponse = false
+    var serviceNameHeader = "x-service-name"
+    var requestedServiceNameHeader = "x-requested-service-name"
```

Comment:
Can it be a misleading name? For example, some requests directed to envoy ingress originate from Load balancers. `authority` header contains domain name, not service name in such cases. Maybe "requested-authority"? Or maybe this header is even not needed at all?

Reply:
This header is needed for debugging. Sometimes it might be problematic to keep trace in application which service was requested. That header might allow to get 2 information at once and just validate. I agree `authority` might be better.